zerver/lib/validator.py

```diff
@@ -228,7 +228,22 @@ def check_url(var_name: str, val: object) -> Optional[str]:
     except ValidationError:
         return _('%s is not a URL') % (var_name,)
 
-def validate_field_data(field_data: ProfileFieldData) -> Optional[str]:
+def check_url_pattern(var_name: str, val: object) -> Optional[str]:
+    error = check_string(var_name, val)
+    if error:
+        return error
+    val = cast(str, val)
+
+    if val.count('%(username)s') != 1:
+        return _('username should appear exactly once in pattern.')
+    val.replace('(username)', '')
```

Comment:
Conceptually, I think we should replace it with just the string `"username"`, in case the URL has a structure where a `/%(username)s/ -> //` would be weird.  (I don't think this is a super likely corner case, but it's more obviously correct that way).